core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -246,6 +249,15 @@ class HadoopRDD[K, V](
         } catch {
           case eof: EOFException =>
             finished = true
+          case e: Exception =>
```

Comment:
will the corrupt files throw something more specific than general Exception? seems a bit too wide of a net to cast here.